pkg/flypg/zombie_test.go

```diff
@@ -0,0 +1,243 @@
+				ExpectedZombie:   false,
+			},
+			// Test case: 5
+			// 3 member setup where both standby's are offline.
+			{
+				MyHostname:       "host-1",
+				TotalMembers:     3,
+				TotalActive:      1,
+				TotalInactive:    2,
+				ConflictMap:      map[string]int{},
+				ExpectedHostname: "",
+				ExpectedZombie:   true,
+			},
+			// Test case: 6
+			// 3 member setup where both standbys agree that i'm not the primary.
+			{
+				MyHostname:    "host-1",
+				TotalMembers:  3,
+				TotalActive:   3,
+				TotalInactive: 0,
+				ConflictMap: map[string]int{
+					"secret-primary": 2,
+				},
+				ExpectedHostname: "secret-primary",
+				ExpectedZombie:   true,
+			},
+			// Test case: 7
+			// 3 member setup where 1 standby disagrees that i'm primary.
+			{
+				MyHostname:    "host-1",
+				TotalMembers:  3,
+				TotalActive:   3,
+				TotalInactive: 0,
+				ConflictMap: map[string]int{
+					"secret-primary": 1,
+				},
+				ExpectedHostname: "host-1",
+				ExpectedZombie:   false,
+			},
+			// Test case: 8
+			// 3 member setup where both standbys report different primaries.
+			{
+				MyHostname:    "host-1",
+				TotalMembers:  3,
+				TotalActive:   3,
+				TotalInactive: 0,
+				ConflictMap: map[string]int{
+					"secret-primary":   1,
+					"secret-primary-2": 1,
+				},
+				ExpectedHostname: "",
+				ExpectedZombie:   true,
+			},
+			// Test case: 9
+			// 4 member setup
+			{
+				MyHostname:       "host-1",
+				TotalMembers:     4,
+				TotalActive:      4,
+				TotalInactive:    0,
+				ConflictMap:      map[string]int{},
+				ExpectedHostname: "host-1",
+				ExpectedZombie:   false,
+			},
+			// Test case: 10
+			// 4 member setup with 1 standby that is inactive
+			{
+				MyHostname:       "host-1",
+				TotalMembers:     4,
+				TotalActive:      3,
+				TotalInactive:    1,
+				ConflictMap:      map[string]int{},
+				ExpectedHostname: "host-1",
+				ExpectedZombie:   false,
+			},
+			// Test case: 11
+			// 4 member setup with 2 standbys that are inactive ( unable to meet quorum)
+			{
+				MyHostname:       "host-1",
+				TotalMembers:     4,
+				TotalActive:      2,
+				TotalInactive:    2,
+				ConflictMap:      map[string]int{},
+				ExpectedHostname: "",
+				ExpectedZombie:   true,
+			},
+			// Test case: 12
+			// 4 member setup with 2 standbys agreeing on a different primary.
+			{
+				MyHostname:    "host-1",
+				TotalMembers:  4,
+				TotalActive:   4,
+				TotalInactive: 0,
+				ConflictMap: map[string]int{
+					"secret-host": 2,
+				},
+				ExpectedHostname: "",
+				ExpectedZombie:   true,
+			},
+			// Test case: 13
+			// 4 member setup with 3 standbys agreeing on a different primary.
+			{
+				MyHostname:    "host-1",
+				TotalMembers:  4,
+				TotalActive:   4,
+				TotalInactive: 0,
+				ConflictMap: map[string]int{
+					"secret-host": 3,
+				},
+				ExpectedHostname: "secret-host",
+				ExpectedZombie:   true,
+			},
+		},
+	}
+
+	for i, c := range tests.Cases {
```

Comment:
I'm not personally a fan of table tests for anything but really simple tests. I find they get really complicated as you start adding edge cases. I'd probably rework these into subtests using `t.Run()`.

Although, if you want to keep the table tests, you can still wrap each of these in a `t.Run()` and then you can filter on them with `go test -run=XXX`